src/dsp_tools/models/xmlresource.py

```diff
@@ -175,7 +188,7 @@ def get_propvals(
 
     def get_bitstream_information(
         self, internal_file_name_bitstream: str, permissions_lookup: dict[str, Permissions]
-    ) -> Optional[dict[str, Union[str, Permissions]]]:
+    ) -> BitstreamInfo | None:
         """
         Get the bitstream object belonging to the resource
```

Comment:
I know this was not written by you, but if you understand what happens here: Could you write a description that is helpful?  Because I don't get it...